src/rampup/OverridingHashcodeAndEqualsMethod.java

```diff
@@ -0,0 +1,42 @@
+package rampup;
+
+import java.util.Objects;
+
+class Test{
+    int roll;
+    String name;
+
+    public Test(int roll, String name) {
+        this.roll = roll;
+        this.name = name;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        Test test=(Test)o;
+        return this.roll == test.roll && this.name.equals((test.name));
+    }
+
+    @Override
+    public int hashCode() {
+        return name.hashCode();
+    }
+
+    @Override
+    public String toString() {
+        return roll+"@"+hashCode();
```

Comment:
This is not the right string implementation. Please Check it online or you can take IDE help to generate `toString` method.